plugins/primus_test/lisp/memcheck-malloc.lisp

```diff
@@ -3,6 +3,9 @@
 (defmethod call (name ptr)
   (when (and ptr (= name 'free)
              (not (= ptr *malloc-zero-sentinel*)))
+    (when (and ptr
```

Comment:
`ptr` is already non-NULL here, see the guard above. Also for `when (not cond)` there is <strike>`unless`</strike> `until cond`. Ooops, it should be, of course, `unless`, not `until` and not even `unit` (as it is documented) :)  Could you please add `unless` in addition to (the now deprecated) `until` to the `init.lisp`? And, please, update the docs correspondingly, and mark `until` as deprecated.

So the code will be now much more readable:
```lisp
(defmethod call (name ptr)
  (when (and ptr (= name 'free)
             (not (= ptr *malloc-zero-sentinel*)))
    (unless (memcheck-is-allocated 'malloc ptr)
      (memcheck-acquire 'malloc ptr 1))
    (memcheck-release 'malloc ptr)))

```

Reply:
Ok, I was really confused yesterday :) `unless` is not `until`. `unless` is a dual for `while`, where `until` is dual `when`. So here are the definitions that we should have,

```lisp
(defmacro when (cnd body)
  "(when CND BODY) if CND evaluates to true, the BODY is evaluated and
   the value of the last expression BODY becomes the value of the
   whole expression. Otherwise, if CND evaluates to false, nil is returned."
  (if cnd (prog body) ()))

(defmacro unless (cnd body)
  "(unless CND BODY) if CND is evaluates to true, then BODY is evaluated and
   the value of the last expression in BODY becomes the value of the
   whole expression. Otherwise, if CND evaluates to false, nil is returned."
  (if (not cnd) () body))

(defmacro until (c b)
  "(until COND BODY) if COND evaluates to false then the whole expression
    evaluates to nil and BODY is not evaluated. Otherwise, BODY is evaluated
    until COND evaluates to true and returns the value of the last evaluation
    of BODY."
  (while (not c) b))
```

Reply:
Ok, I was really confused yesterday :) `unless` is not `until`. `unless` is a dual for `when`, where `until` is dual `while`. So here are the definitions that we should have,
```lisp
(defmacro when (cnd body)
  "(when CND BODY) if CND evaluates to true, then BODY is evaluated and
   the value of the last expression in BODY becomes the value of the
   whole expression. Otherwise, if CND evaluates to false, nil is returned."
  (if cnd (prog body) ()))

(defmacro unless (cnd body)
  "(unless CND BODY) if CND evaluates to false, then BODY is evaluated and
   the value of the last expression in BODY becomes the value of the
   whole expression. Otherwise, if CND evaluates to true, nil is returned."
  (if (not cnd) () body))

(defmacro until (c b)
  "(until COND BODY) if COND evaluates to true, then the whole expression
   evaluates to nil and BODY is not evaluated. Otherwise, if COND evaluates
   to false, then BODY is evaluated until COND evaluates to true and the value
   of the last evaluation of BODY becomes the value of the whole expression."
  (while (not c) b))
```